### PR TITLE
[WIP] CMake: SENSEI needs AMReX_AMRLEVEL

### PR DESCRIPTION
**Pull request by ax3l**
Internal cross-dependency on a functionality that we usually do not need in AMReX.

X-ref: https://github.com/ECP-WarpX/WarpX/pull/2192#issuecomment-901285722

### CODE REVIEW

**Comment by c-wetterer-nelson** [comment]
I wonder if this would be better done upstream in AMReX? No need to add the AmrLevel dependency unnecessarily here. The SENSEI bridge we used here ought not to be including that dependency, the problem is that the particular bridge that DOES need that dependency is under the same compiler option umbrella as the bridge we want to use in WarpX. That "problem" bridge is being brought in without its AMReX dependency, but really it should be turned off altogether if AmrLevel is not being brought in.

File: cmake/dependencies/AMReX.cmake [hunk 1]
```diff
@@ -63,10 +63,12 @@ macro(find_amrex)
 
         if(WarpX_SENSEI)
             set(AMReX_SENSEI ON CACHE INTERNAL "")
+            set(AMReX_AMRLEVEL ON CACHE INTERNAL "")
```
Comment (ax3l) [inline cmake/dependencies/AMReX.cmake]: Add fixme comment:
```suggestion
            set(AMReX_AMRLEVEL ON CACHE INTERNAL "")  # FIXME in AMReX
```

**Review by ax3l — COMMENTED** [review]
Fully agree, let's fix the upstream issue.
Will open an issue to track it.

I would merge this work-around for now so it builds and we can soon test CI and Spack with it.

**Comment by ax3l** [comment]
Documented follow-up work upstream in https://github.com/AMReX-Codes/amrex/issues/2263 :)

**Comment by c-wetterer-nelson** [comment]
https://github.com/AMReX-Codes/amrex/pull/2258 upstream fix

**Comment by ax3l** [comment]
Replaced by #2235 and #2236

**Pull request closed without merging**